src/datasets/domain/useCases/GetDataset.ts

```diff
@@ -0,0 +1,15 @@
+import { UseCase } from '../../../core/domain/useCases/UseCase';
+import { IDatasetsRepository } from '../repositories/IDatasetsRepository';
+import { Dataset } from '../models/Dataset';
+
+export class GetDataset implements UseCase<Dataset> {
+  private datasetsRepository: IDatasetsRepository;
+
+  constructor(datasetsRepository: IDatasetsRepository) {
+    this.datasetsRepository = datasetsRepository;
+  }
+
+  async execute(datasetId?: number, datasetPersistentId?: string, datasetVersionId?: number): Promise<Dataset> {
```

Comment:
Can we search at the same time by id + persistentId?

If not, we could provide 2 different execute functions, 1 for the id and 1 for the persistenId

With an overload of the execute() or with different use cases

Reply:
Typescript doesn't allow me to do method overloading of the _execute_ function.

I have created two different use cases, to comply with the UseCase interface.